### ISSUE TITLE
Using Halide profiler on Windows with AOT

### ISSUE THREAD

**Issue by 9cvele3** [halide/Halide#3198]
I have developed a few filters in Halide. I am deploying them like proposed in [example 15 in tutorials](http://halide-lang.org/tutorials/tutorial_lesson_15_generators_usage.html). So, I generate runtime only once in order to avoid linker errors on Windows. 

I would like to be able to use Halide profiler on new filters that I add. If I compile all the filters I have with profile option, then I have linker errors for multiple defined symbols. If I add profile option to just one filter that I am trying to profile, there is no linker errors, but when running generated function, nothing is displayed. I also tried generating runtime with profile option, but without results.

I am able to use Halide profiler with JIT without problems.

**Comment by abadams** [comment]
Which symbols were multiply-defined? It's possible we have a wrong linkage type on something.

**Comment by 9cvele3** [comment]
These symbols are causing problems:

- _halide_profiler_set_current_func_
- _halide_profiler_incr_active_threads_
- _halide_profiler_decr_active_threads_

**Comment by 9cvele3** [comment]
@abadams 
All the symbols that are causing problems are in _src/runtime/profiler_inlined.cpp_. Is weak linking necessary for them ?

**Comment by abadams** [comment]
I don't think so. I think that's a mistake. They should be always_inline but not WEAK.
